actionview/lib/action_view/template/error.rb

```diff
@@ -93,7 +94,11 @@ def initialize(error)
       def corrections
         path = @error.path
         prefixes = @error.prefixes
-        candidates = @error.paths.flat_map(&:all_template_paths).uniq
+        if @error.partial
+          candidates = @error.paths.flat_map(&:all_template_paths).uniq.filter { |path| path.match?(%r{_([^/]+)$}) }
+        else
+          candidates = @error.paths.flat_map(&:all_template_paths).uniq.filter { |path| path.match?(%r{([^/]+)$}) }
+        end
```

Comment:
@matthewd is this what you imagined only searching partials if you want a partial would look like?

Reply:
This should speedup missed template search since we have to calculate the distance on less template names, which is a nice side effect.